From d1f77b3f0c89f0fc8328298a1012bdc96e352f76 Mon Sep 17 00:00:00 2001
From: Kent Russell <kent.russell@amd.com>
Date: Tue, 21 Nov 2017 08:44:57 -0500
Subject: [PATCH 2864/4131] drm/amdkfd: Fix PTR_ERR errors

If the variable is 0, we're passing 0 to PTR_ERR, which is useless

Change-Id: I132d1dd638b23617445e723b3d645ea031dc91ca
Signed-off-by: Kent Russell <kent.russell@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_chardev.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_chardev.c b/drivers/gpu/drm/amd/amdkfd/kfd_chardev.c
index 77cd6b2..f99fc4e 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_chardev.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_chardev.c
@@ -1296,7 +1296,7 @@ static int kfd_ioctl_map_memory_to_gpu(struct file *filep,
 	mem = kfd_process_device_translate_handle(pdd,
 						GET_IDR_HANDLE(args->handle));
 	if (!mem) {
-		err = PTR_ERR(mem);
+		err = -ENOMEM;
 		goto get_mem_obj_from_handle_failed;
 	}
 
@@ -1421,14 +1421,14 @@ static int kfd_ioctl_unmap_memory_from_gpu(struct file *filep,
 	pdd = kfd_get_process_device_data(dev, p);
 	if (!pdd) {
 		pr_debug("Process device data doesn't exist\n");
-		err = PTR_ERR(pdd);
+		err = -ENODEV;
 		goto bind_process_to_device_failed;
 	}
 
 	mem = kfd_process_device_translate_handle(pdd,
 						GET_IDR_HANDLE(args->handle));
 	if (!mem) {
-		err = PTR_ERR(mem);
+		err = -ENOMEM;
 		goto get_mem_obj_from_handle_failed;
 	}
 
-- 
2.7.4

